packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -38,19 +38,19 @@ import { getAllModels, getSdkEnum, getSdkModel } from "./types.js";
 export const namespace = "Azure.ClientGenerator.Core";
 
 function getScopedDecoratorData(context: SdkContext, key: symbol, target: Type): any {
-  const retval = context.program.stateMap(key).get(target);
+  const retval: Record<string, any> = context.program.stateMap(key).get(target);
   if (retval === undefined) return retval;
-  if (!retval.scopes) return retval.value; // this means it applies to all languages
-  if (retval.scopes.includes(context.emitterName)) return retval.value;
-  return undefined;
+  if (Object.keys(retval).includes(context.emitterName)) return retval[context.emitterName];
+  return retval[""]; // in this case it applies to all languages
```

Comment:
nit: any chance we could use a value to determine all instead of empty string? `*` maybe?

Reply:
or a `symbol` to prevent any conflicts

Reply:
yep will do